salalib/linkutils.cpp

```diff
@@ -72,4 +72,29 @@ namespace depthmapX {
         }
         return mergePixelPairs;
     }
+
+    void mergePixelPairs(std::vector<PixelRefPair>& links, PointMap& currentMap) {
+
+        // check if any link pixel already exists on the map
+        for (size_t i = 0; i < links.size(); i++)
+        {
+            PixelRefPair link = links[i];
+            if(currentMap.isPixelMerged(link.a)
+                    || currentMap.isPixelMerged(link.b))
+            {
+                // one of the cells is already on the map
+                std::stringstream message;
+                message << "Link pixel found that is already linked on the map "
+                        << "(line " << i << ")"
+                        << flush;
+                throw depthmapX::RuntimeException(message.str().c_str());
```

Comment:
Hang on, so why is it an error if a pixel is already linked/merge, but not it appears twice in the list of new links? If there's no reason for the split, the two loops could be one, couldn't they?